src/pybind/mgr/prometheus/module.py

```diff
@@ -432,15 +439,54 @@ def _setup_static_metrics(self):
                 'Number of {} objects'.format(state),
             )
 
+        for check in HEALTH_CHECKS:
+            path = 'healthcheck_{}'.format(check['name'].lower())
+            metrics[path] = Metric(
+                'gauge',
+                path,
+                check['description'],
+            )
+
         return metrics
 
     @profile_method()
     def get_health(self):
+
+        def _get_value(message, delim=" ", word_pos=0):
+            """Extract value from message (default is 1st field)"""
+            v_str = message.split(delim)[word_pos]
```

Comment:
the "message" here was designed to be a human readable message and is formatted like

```
{num} slow ops, oldest one blocked for {secs} sec, daemons {daemons_list[:10]} have slow ops.
```

since there are multiple values aggregated for a single metric, for instance, the daemons which have slow requests, and the total number of slow request. it's difficult to use a single number to represent them.

as a compromise, could you put a sample of the message here, and instead of using `str.split()`, probably use regexp to capture the number? like

```py
# parse the human-readable summary like
# 42 slow ops, oldest one blocked for 12 sec, daemons [osd.0, osd.3] have slow ops.
matched = re.match(r'(?P<num>\d+) slow ops', summary)
if matched:
  return int(matched.group('num'))
else:
  raise ValueError(f'bad format of SLOW_OPS:summary: "{summary}"')
```

Reply:
The idea was to enable further healthcheck based metrics to be added later, so I kept the get_value as generic as possible instead of opting for a healthcheck specific regex/function.

Reply:
what i don't like is to use `split` to parse a human readable string. this string is designed to be a space separated values.

Reply:
sorry - I don't follow you. I'm splitting, so for another alert the value to extract could be at the end, so with split I can just sepect workpos -1 - and not need separate regex's for every potential healthcheck that we would want to add.

Reply:
anyway, i don't strong in this. but to me, it's like parsing an XML using regexp or split. i would parse a user facing string using regexp instead relying on `split` and an `0` index (i.e. a magic number).